src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -51,6 +53,7 @@ internal class InProcLanguageServer
             CodeAnalysis.Workspace workspace,
             IDiagnosticService diagnosticService,
             IAsynchronousOperationListenerProvider listenerProvider,
+            ILspSolutionProvider solutionProvider,
```

Comment:
Now that this is being passed, can we remove the 'workspace' parameter in favor of using the main workspace as handed through the ILspSolutionProvider?

Reply:
Or put a different way: what happens if a workspace other than the main workspace from ILspSolutionProvider is passed? Is that bad?

Reply:
At one stage I wanted to get rid of the ILspSolutionProvider altogether, and I wouldn't rule it out in the end. As you note there is an odd relationship here.